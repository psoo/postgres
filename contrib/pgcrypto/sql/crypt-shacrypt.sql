--
-- crypt() and gensalt: sha256crypt, sha512crypt
--

-- $5$ is sha256crypt
SELECT crypt('', '$5$Szzz0yzz');

SELECT crypt('foox', '$5$Szzz0yzz');

CREATE TABLE ctest (data text, res text, salt text);
INSERT INTO ctest VALUES ('password', '', '');

-- generate a salt for sha256crypt, default rounds
UPDATE ctest SET salt = gen_salt('sha256crypt');
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;

-- generate a salt for sha256crypt, rounds 9999
UPDATE ctest SET salt = gen_salt('sha256crypt', 9999);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;

TRUNCATE ctest;

-- $6$ is sha512crypt
SELECT crypt('', '$6$Szzz0yzz');

SELECT crypt('foox', '$6$Szzz0yzz');

-- generate a salt for sha512crypt, default rounds
UPDATE ctest SET salt = gen_salt('sha512crypt');
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;

-- generate a salt for sha512crypt, rounds 9999
UPDATE ctest SET salt = gen_salt('sha512crypt', 9999);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;

-- cleanup
DROP TABLE ctest;