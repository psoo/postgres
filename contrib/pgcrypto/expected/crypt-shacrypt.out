--
-- crypt() and gensalt: sha256crypt, sha512crypt
--
-- $5$ is sha256crypt
SELECT crypt('', '$5$Szzz0yzz');
                          crypt                          
---------------------------------------------------------
 $5$Szzz0yzz$cA.ZFZKqblRYjdsbrWtVTYa/qSwPQnt2uh0LBtyYAAD
(1 row)

SELECT crypt('foox', '$5$Szzz0yzz');
                          crypt                          
---------------------------------------------------------
 $5$Szzz0yzz$7hI0rUWkO2QdBkzamh.vP.MIPlbZiwSvu2smhSi6064
(1 row)

CREATE TABLE ctest (data text, res text, salt text);
INSERT INTO ctest VALUES ('password', '', '');
-- generate a salt for sha256crypt, default rounds
UPDATE ctest SET salt = gen_salt('sha256crypt');
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
 worked 
--------
 t
(1 row)

-- generate a salt for sha256crypt, rounds 9999
UPDATE ctest SET salt = gen_salt('sha256crypt', 9999);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
 worked 
--------
 t
(1 row)

TRUNCATE ctest;
-- $6$ is sha512crypt
SELECT crypt('', '$6$Szzz0yzz');
                                               crypt                                                
----------------------------------------------------------------------------------------------------
 $6$Szzz0yzz$EGj.JLAovFyAtCJx3YD1DXD1yTXoO9gv4qgLyHBsJJ1lkpnLB8ZPHekm1qXjJCOBc/8thCuHpxNN8Y5xzRYU5.
(1 row)

SELECT crypt('foox', '$6$Szzz0yzz');
                                               crypt                                                
----------------------------------------------------------------------------------------------------
 $6$Szzz0yzz$KqDw1Y8kze.VFapkvTc9Y5fbqzltjeRz1aPGC/pkHRhFQZ2aM6PmZpXQjcD7AOH88Bq0CSD.VlmymQzcBMEUl0
(1 row)

INSERT INTO ctest VALUES ('password', '', '');
-- generate a salt for sha512crypt, default rounds
UPDATE ctest SET salt = gen_salt('sha512crypt');
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
 worked 
--------
 t
(1 row)

-- generate a salt for sha512crypt, rounds 9999
UPDATE ctest SET salt = gen_salt('sha512crypt', 9999);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
 worked 
--------
 t
(1 row)

-- cleanup
DROP TABLE ctest;
